src/record_syscall.cc

```diff
@@ -3783,8 +3783,9 @@ static void process_mremap(RecordTask* t, remote_ptr<void> old_addr,
     return;
   }
 
-  // Now record the new part of the mapping.
-  km = m.map.subrange(new_addr + old_size, new_addr + new_size);
+  // Now record the new part of the mapping, possibly with some overlap.
+  km = m.map.subrange(new_addr + floor_page_size(old_length),
```

Comment:
In your testcase the file size is `2 * num_bytes` which looks like it should work with the current code.